### ISSUE TITLE
cloud-azure plugin not working 

### ISSUE THREAD

**Issue by hisuwh** [elastic/elasticsearch#15630]
I get the following error which takes down my es service:

```
java.lang.RuntimeException: Service or property not registered:  com.microsoft.windowsazure.management.compute.ComputeManagementClient interface com.microsoft.windowsazure.management.compute.ComputeManagementClient
	at com.microsoft.windowsazure.core.DefaultBuilder.build(DefaultBuilder.java:197)
	at com.microsoft.windowsazure.Configuration.create(Configuration.java:113)
	at com.microsoft.windowsazure.management.compute.ComputeManagementService.create(ComputeManagementService.java:47)
	at org.elasticsearch.cloud.azure.management.AzureComputeServiceImpl.<init>(AzureComputeServiceImpl.java:83)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at <<<guice>>>
	at org.elasticsearch.node.Node.<init>(Node.java:198)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:170)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:270)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
```

Elasticsearch: 2.0
Java: 1.8.0_66
Windows Server 2012 R2 Datacenter

**Comment by hisuwh** [comment]
Have tried with Java 7 to no avail

**Comment by dadoonet** [comment]
Can you share your settings?

**Comment by hisuwh** [comment]
my elasticsearch settings?

```
cluster.name: empactislive

network.bind_host: _en0_
network.publish_host: _en0_

cloud.azure.management.keystore.path: E:\azurekeystore.pkcs12
cloud.azure.management.keystore.type: pkcs12
cloud.azure.management.keystore.password: ***********
cloud.azure.management.subscription.id: *******-****-****-****-********
cloud.azure.management.cloud.service.name: empctslivees
discovery.azure.deployment.name: empctsesvmlive1
discovery.type: azure
```

**Comment by hisuwh** [comment]
I get the same error with elasticsearch 2.1.1

**Comment by dadoonet** [comment]
Can you try \\ instead of \?

Anything else in logs?

**Comment by hisuwh** [comment]
Do you mean / instead of \ ?

I can't see anything else in the logs. 
I have tried changing the discovery logging level to trace but that does not give any more information

**Comment by hisuwh** [comment]
Tried both 
```
\\
```
and 
```
/
```
to no avail

**Comment by hisuwh** [comment]
I am also a lot of this warning:
```
Java HotSpot(TM) 64-Bit Server VM warning: Using the ParNew young collector with the Serial old collector is deprecated and will likely be removed in a future release
Exception in thread "main" puteManagementClient
	at com.microsoft.windowsazure.core.DefaultBuilder.build(DefaultBuilder.java:197)
	at com.microsoft.windowsazure.Configuration.create(Configuration.java:113)
	at com.microsoft.windowsazure.management.compute.ComputeManagementService.create(ComputeManagementService.java:47)
	at org.elasticsearch.cloud.azure.management.AzureComputeServiceImpl.<init>(AzureComputeServiceImpl.java:83)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at <<<guice>>>
	at org.elasticsearch.node.Node.<init>(Node.java:198)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:170)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:270)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
```

**Comment by david-mills** [comment]
im getting the same issue with 2.1.1, and the same confguration as above, not sure if its to do with the fact i installed the 'cloud-azure' plugin instead of 'discovery-azure' as it says in the latest docs.  discovery-azure gave me a file not found exception as the zip doesnt exist

**Comment by dadoonet** [comment]
@hisuwh Just to make sure about the source of your issue, can you move the pkcs12 file on the same drive as you have elasticsearch instance (I guess it could be for example `C:`) and then change your `elasticsearch.yml` accordingly:

```yaml
cloud.azure.management.keystore.path: /azurekeystore.pkcs12
```

**Comment by hisuwh** [comment]
I have been away for Christmas as you might expect. I will try this out at some point and let you know if this helps

**Comment by hisuwh** [comment]
I have tried moving my keystore to the c drive but this makes no difference

**Comment by dadoonet** [comment]
@hisuwh And your `elasticsearch.yml` is now?

```yml
cloud.azure.management.keystore.path: /azurekeystore.pkcs12
```

**Comment by hisuwh** [comment]
yes

**Comment by hisuwh** [comment]
I am sorry for the bump.
But any more ideas on this?
This massively blocking us at the moment.
Has anyone your end tried to reproduce this? I don't know if its a bug with your tools or in my setup.
I have the same setup on 3 other clusters except they are on older versions of elasticsearch.

**Comment by georgievrado** [comment]
I am having the same problem on elasticsearch 2.1.1 (running on ubuntu 15.04 installed from repositories). My config is as follows:
```yaml
cluster.name: "mycluster"
bootstrap.mlockall: true
path:
 logs: /var/log/elasticsearch
 data: /datadrive/elasticsearch/data
cloud:
 azure:
  management:
   keystore:
    path: /etc/elasticsearch/mykeystorefile.pkcs12
    password: password
   subscription.id: GUID
   cloud.service.name: myservicename
  storage:
   account: mystorageaccount
   key: mystorageaccountkey
discovery:
 type: azure
refresh_interval: 30s
```

**Comment by hisuwh** [comment]
@dadoonet - any more thoughts? We are currently running a 1 node cluster in live because of this which presents a pretty huge risk

**Comment by georgievrado** [comment]
@hisuwh I've worked around by disabling azure discovery and relying on unicast, by hardcoding the nodes.

**Comment by hisuwh** [comment]
Of course, that is a good point.  Thanks
Good workaround though by no means a solution

**Comment by clintongormley** [comment]
@gmarz are you familiar with the azure plugin? Is this something you could investigate?

**Comment by dadoonet** [comment]
@hisuwh @georgievrado Any chance you could turn on `TRACE` level? 

In your `logging.yml`, add:

```yml
  org.elasticsearch.cloud.azure: TRACE
  org.elasticsearch.discovery.azure: TRACE
```

It might give us some ideas about what is happening. Thanks!

**Comment by tlrx** [comment]
This issue is closed in c557663b90dc3ac3f0ae7391cb6eac543812a946

Please note that since ES 2.0, the keystore used by the Azure plugin must be placed in a directory accessible by the elasticsearch process. I suggest to place it in the `config` directory.
